src/btree/bt_split.c

```diff
@@ -555,9 +567,11 @@ __split_deepen(WT_SESSION_IMPL *session, WT_PAGE *parent, uint32_t children)
 		if (!WT_PAGE_IS_INTERNAL(child))
 			continue;
 #ifdef HAVE_DIAGNOSTIC
-		__split_verify_intl_key_order(session, child);
+		WT_WITH_PAGE_INDEX(session,
+		    __split_verify_intl_key_order(session, child));
 #endif
-		WT_INTL_FOREACH_BEGIN_SAFE(session, child, child_ref) {
+		WT_ENTER_PAGE_INDEX(session);
+		WT_INTL_FOREACH_BEGIN(session, child, child_ref) {
```

Comment:
This is the bug fix part of this change, right? Could you add a (big) comment explaining why we need to have split_gen set here?